### ISSUE TITLE
[Bug] OBS virtual camera not listed in settings

### ISSUE THREAD

**Issue by the-arkhive** [Vencord/Vesktop#574]
### Operating System

Arch

### Linux Only ~ Desktop Environment

Wayland

### What happens when the bug or crash occurs?

OBS virtual camera does not appear in video options. There is an external USB webcam that does get listed as expected

### What is the expected behaviour?

Video device should appear in the drop down menu.

### How do you recreate this bug or crash?

1: Install latest version of Vesktop
2: Start OBS virtual camera
3: Go to "Voice & Video" settings
4: select drop down for video devices

### Debug Logs

There would not seem to be any relevant logs printed to the console when selecting the drop down menu. This works as intended in the discord-canary installed from the official Arch 

### Request Agreement

- [X] I have searched the existing issues and found no similar issue
- [X] I am using the latest Vesktop and Vencord versions
- [X] This issue occurs on an official release (not just the AUR or Nix packages)

**Comment by Blank-0104** [comment]
Hey I'm using Hyprland (wayland) on gentoo, and I cant get droidcam to show up either, no clue with physical cameras as I dont have one but it does still show up on vencord. I don't know if it's related but I think it might be

**Comment by PolisanTheEasyNick** [comment]
I have interesting info. I have installed Iriun Webcam application. But even if I do not start Iriun Webcam but start OBS Virtual Camera then I will have in discord OBS camera available but with name "Iriun Webcam". Maybe there is something wrong with V4L?

**Comment by jwe66** [comment]
I also am experiencing the exact same bug and explained in the original post. I am also getting this issue with discord web client however. Outside of discord, my system does recognize the OBS Camera

**Comment by Mikilio** [comment]
As far as I know, this never actually worked on discord ever. So, this is not a bug but a feature request.

**Comment by LeviOP** [comment]
@Mikilio As I alluded to in my previous comment, I can get this to work fine on Vesktop (and presumably stock discord, as nothing about the camera access changes). If it's not working for you, could you answer the questions I asked in my last comment?

**Comment by Mikilio** [comment]
Interesting, that it works for you. It never did for me on discord only. I load the module on boot at stage 2 by specifying it on NixOS which uses systemd-module-load under the hood.

**Comment by LeviOP** [comment]
To be clear, you're you're specifying `exclusive_caps=1` to v4l2loopback when you load the module in your nix config, right? It will not work without this. Also, can you get the obs virtual camera to work in any application / browser?

**Comment by Mikilio** [comment]
I did not specify that parameter, but it did work on all applications I use except for discord. I will tay that right now.

**Comment by LeviOP** [comment]
Yes. The problem _is_ with chromium / the outdated version of electron that Discord and Vesktop use not correctly implementing camera handling, which requires exclusive caps as a fix. You can read the link in my initial comment for a more detailed explanation.

**Comment by Mikilio** [comment]
Can confirm it works. For any other NixOS users:
```nix
{
  config,
  ...
}:
{
  boot = {
    extraModulePackages = with config.boot.kernelPackages; [
      v4l2loopback.out
    ];
    kernelModules = [ "v4l2loopback" ];
    extraModprobeConfig = ''
      options v4l2loopback exclusive_caps=1
    '';
  };
}
```

**Comment by LeviOP** [comment]
@Vendicated This is either user error or could be considered an upstream issue with electron / chromium. Probably should be closed
